src/cascadia/TerminalSettingsModel/Theme.cpp

```diff
@@ -0,0 +1,339 @@
+    return *result;
+}
+
+winrt::Microsoft::Terminal::Settings::Model::ThemeColor ThemeColor::FromAccent() noexcept
+{
+    auto result = winrt::make_self<implementation::ThemeColor>();
+    result->_ColorType = ThemeColorType::Accent;
+    return *result;
+}
+
+winrt::Microsoft::Terminal::Settings::Model::ThemeColor ThemeColor::FromTerminalBackground() noexcept
+{
+    auto result = winrt::make_self<implementation::ThemeColor>();
+    result->_ColorType = ThemeColorType::TerminalBackground;
+    return *result;
+}
+
+static wil::unique_hkey openDwmRegKey()
+{
+    HKEY hKey{ nullptr };
+    if (RegOpenKeyEx(HKEY_CURRENT_USER, RegKeyDwm, 0, KEY_READ, &hKey) == ERROR_SUCCESS)
+    {
+        return wil::unique_hkey{ hKey };
+    }
+    return nullptr;
+}
+static DWORD readDwmSubValue(const wil::unique_hkey& dwmRootKey, const wchar_t* key)
+{
+    DWORD val{ 0 };
+    DWORD size{ sizeof(val) };
+    LOG_IF_FAILED(RegQueryValueExW(dwmRootKey.get(), key, nullptr, nullptr, reinterpret_cast<BYTE*>(&val), &size));
+    return val;
+}
+
+static til::color _getAccentColorForTitlebar()
+{
+    // The color used for the "Use Accent color in the title bar" in DWM is
+    // stored in HKCU\Software\Microsoft\Windows\DWM\AccentColor.
+    return til::color{ static_cast<COLORREF>(readDwmSubValue(openDwmRegKey(), RegKeyAccentColor)) };
+}
+
+til::color ThemeColor::ColorFromBrush(const winrt::Windows::UI::Xaml::Media::Brush& brush)
+{
+    if (auto acrylic = brush.try_as<winrt::Windows::UI::Xaml::Media::AcrylicBrush>())
+    {
+        return acrylic.TintColor();
+    }
+    else if (auto solidColor = brush.try_as<winrt::Windows::UI::Xaml::Media::SolidColorBrush>())
+    {
+        return solidColor.Color();
+    }
+    return {};
+}
+
+winrt::Windows::UI::Xaml::Media::Brush ThemeColor::Evaluate(const winrt::Windows::UI::Xaml::ResourceDictionary& res,
+                                                            const winrt::Windows::UI::Xaml::Media::Brush& terminalBackground,
+                                                            const bool forTitlebar)
+{
+    static const auto accentColorKey{ winrt::box_value(L"SystemAccentColor") };
+
+    // NOTE: Currently, the DWM titlebar is always drawn, underneath our XAML
+    // content. If the opacity is <1.0, the you'll be able to see it, including
+    // the original caption buttons, which we don't want.
+
+    switch (ColorType())
+    {
+    case ThemeColorType::Accent:
+    {
+        til::color accentColor;
+        if (forTitlebar)
+        {
+            accentColor = _getAccentColorForTitlebar();
+        }
+        else
+        {
+            accentColor = winrt::unbox_value<winrt::Windows::UI::Color>(res.Lookup(accentColorKey));
+        }
+
+        const auto accentBrush = winrt::Windows::UI::Xaml::Media::SolidColorBrush();
+        accentBrush.Color(accentColor);
+        if (forTitlebar)
+        {
+            accentBrush.Opacity(1.0);
+        }

```

Comment:
Couldn't we just set the alpha to 255 for `_getAccentColorForTitlebar`'s return value?